src/smt/smt_engine.cpp

```diff
@@ -2535,8 +2580,11 @@ void SmtEngine::checkUnsatCore() {
   Notice() << "SmtEngine::checkUnsatCore(): generating unsat core" << endl;
   UnsatCore core = getUnsatCore();
 
-  SmtEngine coreChecker(d_exprManager);
+  SmtEngine coreChecker(d_exprManager, &d_options);
+  coreChecker.setIsInternalSubsolver();
   coreChecker.setLogic(getLogicInfo());
+  coreChecker.setOption("check-unsat-cores", false);
```

Comment:
Nit: Why use the external way of setting options here instead of `coreChecker.getOptions()[options::checkUnsatCores]`?

Reply:
Sure I can change this.

Reply:
What exactly are you suggesting to change this to?
We could change to `coreChecker.getOptions().setOption("check-unsat-cores", "false");` I guess?

Reply:
Sorry, I wrote something wrong above. I meant `coreChecker.getOptions().set(options::checkUnsatCores, false)`.

Reply:
Updated this.